crates/prompt_gateway/src/prompt_filter_context.rs

```diff
@@ -285,20 +279,14 @@ impl RootContext for FilterContext {
             GatewayMode::Llm => None,
             GatewayMode::Prompt => Some(Rc::clone(self.embeddings_store.as_ref().unwrap())),
```

Comment:
I have pointed this out a couple times. This is a bug. The point here is to check if the embedding store is present (i.e fully initialized), otherwise this function `create_http_context` should early return here with `None`

Reply:
took your change from https://github.com/katanemo/arch/pull/183